src/libraries/System.IO.Compression/src/System/IO/Compression/DeflateZLib/DeflateStream.cs

```diff
@@ -281,7 +281,14 @@ internal int ReadCore(Span<byte> buffer)
                     int n = _stream.Read(_buffer, 0, _buffer.Length);
                     if (n <= 0)
                     {
-                        break;
+                        if (!_inflater.Finished() && _inflater.NeedsInput() && _inflater.AvailableOutput != 1)
```

Comment:
` _inflater.AvailableOutput != 1` feels awkward. Without it, `Read_Eof_Returns` fails.